src/test/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoServiceSpec.scala

```diff
@@ -18,31 +22,76 @@ class LeonardoServiceSpec extends TestKit(ActorSystem("leonardotest")) with Flat
     super.afterAll()
   }
 
-  val dataprocConfig = ConfigFactory.load().as[DataprocConfig]("dataproc")
+  private val dataprocConfig = ConfigFactory.load().as[DataprocConfig]("dataproc")
+
+  private val gdDAO = new MockGoogleDataprocDAO(dataprocConfig)
+  private val leo = new LeonardoService(dataprocConfig, gdDAO, DbSingleton.ref)
+
+  private val bucketPath = "bucket-path"
+  private val serviceAccount = "service-account"
+  private val googleProject = "test-google-project"
+  private val testClusterRequest = ClusterRequest(bucketPath, serviceAccount, Map.empty)
+
+  val initFiles = Array( dataprocConfig.clusterDockerComposeName, dataprocConfig.initActionsScriptName, dataprocConfig.jupyterServerCrtName,
+    dataprocConfig.jupyterServerKeyName, dataprocConfig.jupyterRootCaPemName, dataprocConfig.jupyterProxySiteConfName)
+
+  "LeonardoService" should "create a cluster" in isolatedDbTest {
+    // set unique names for our cluster
+    val clusterName = s"cluster-${UUID.randomUUID.toString}"
+    val googleProject = s"project-${UUID.randomUUID.toString}"
+
+    // create the cluster
+    val clusterCreateResponse = leo.createCluster(googleProject, clusterName, testClusterRequest).futureValue
 
-  val gdDAO = new MockGoogleDataprocDAO
-  val leo = new LeonardoService(gdDAO, DbSingleton.ref)
+    // check the create response has the correct info
+    clusterCreateResponse.googleBucket shouldEqual bucketPath
+    clusterCreateResponse.googleServiceAccount shouldEqual serviceAccount
+
+    // check the firewall rule was created for the project
+    assert(gdDAO.firewallRules.exists(_  == (googleProject, dataprocConfig.clusterFirewallRuleName)))
```

Comment:
you might be able to do `gdDAO.firewallRules should contain((googleProject, dataprocConfig.clusterFirewallRuleName))`